### PR TITLE
Performance Optimization: Resource Loading & Bundle Size Improvements

### PR DESCRIPTION
**Pull request by namedotget**
# Performance Optimization: Resource Loading & Bundle Size Improvements

## Overview
This PR implements comprehensive performance optimizations focused on reducing bundle size, improving resource loading efficiency, and enhancing runtime performance across the MoonDAO UI.

### 1. Webpack Bundle Optimization
- **Intelligent Code Splitting**: Implemented strategic chunk splitting for major dependencies
  - Separate chunks for `thirdweb`, `privy`, `web3` libraries, and `globe` components
  - Optimized vendor chunk configuration
  - Async loading for heavy 3D components
- **Bundle Analyzer Integration**: Added `@next/bundle-analyzer` for ongoing bundle monitoring
  - Run `npm run analyze` to generate bundle analysis
- **Production Optimizations**:
  - Enabled SWC minification
  - Console statement removal in production (preserving errors/warnings)
  - Disabled `X-Powered-By` header

### 2. Lazy Loading & Code Splitting
- **Globe Components**: Created lazy-loaded wrappers for Earth and Moon 3D components
  - `LazyEarth.tsx` and `LazyMoon.tsx` with intersection observer
  - Prevents loading heavy Three.js dependencies until needed
  - Includes loading states with styled placeholders
- **Component-Level Optimizations**:
  - Added lazy loading to `IPFSRenderer`, `ProgressBar`, `SkeletonLoader`, `Tooltip`
  - Implemented visibility-based rendering with Intersection Observer
  - Dynamic imports for non-critical components

### 3. Animation & Rendering Performance
- **SpaceBackground Optimizations**:
  - Reduced star count by 50% on mobile devices
  - Page Visibility API integration - pauses animations when tab is hidden
  - Optimized `will-change` usage - only applied when page is visible
  - Added CSS containment (`contain: layout style paint`)
  - RequestAnimationFrame optimization for scroll handling
- **CardStack Improvements**:
  - Visibility-based animation pausing
  - Optimized re-render logic with `useMemo`

### 4. Performance Monitoring
- **Web Vitals Tracking**: New `/lib/performance/webVitals.ts` module
  - Tracks Core Web Vitals (LCP, FID, CLS, FCP, TTFB)
  - Long task monitoring (>50ms)
  - Page visibility tracking for battery optimization
  - Connection speed detection
- **Analytics Endpoint**: New `/api/vitals.ts` for collecting performance metrics
- **Integrated in `_app.tsx`**: Automatic monitoring across all pages

### 5. Next.js Configuration Enhancements
- **Image Optimization**: Added AVIF and WebP format support
- **Package Import Optimization**: Configured for `@heroicons/react`, `gsap`, `react-globe.gl`
- **Server Components**: External package configuration for thirdweb

### 6. React Performance Patterns
- **Memoization**: Added `useMemo` for expensive context values in `_app.tsx`
- **Reduced Re-renders**: Optimized context provider value creation
- **Conditional Rendering**: Smart loading states and skeleton screens

## Expected Performance Improvements
- **Bundle Size**: 20-30% reduction in initial JavaScript bundle
- **Time to Interactive**: Faster due to code splitting and lazy loading
- **Mobile Performance**: 50% fewer stars, paused animations when hidden
- **Battery Life**: Reduced CPU usage when tab is not visible
- **3D Components**: Only loaded when user scrolls to them

---

### General Performance
- [ ] Run `yarn analyze` and verify bundle sizes are reasonable
- [ ] Check browser DevTools Network tab - verify lazy loading of chunks
- [ ] Verify initial page load is faster than previous version
- [ ] Check Lighthouse scores (Performance, Accessibility, Best Practices, SEO)

### Homepage
- [ ] Homepage loads without errors
- [ ] SpaceBackground renders correctly with stars
- [ ] Star animations are smooth (60fps)
- [ ] SignedInDashboard renders properly
- [ ] All cards and content load correctly

### Mobile Testing (Viewport < 768px)
- [ ] Homepage renders with reduced star count
- [ ] Animations are smooth on mobile devices
- [ ] No layout shifts or jank
- [ ] Touch interactions work properly
- [ ] Globe components load correctly on mobile

### Map Page
- [ ] Navigate to `/map`
- [ ] Earth globe loads with loading placeholder
- [ ] Globe renders correctly after loading
- [ ] Moon globe (if present) loads correctly
- [ ] 3D interactions work smoothly
- [ ] No console errors related to Three.js

### Animation Performance
- [ ] Open homepage and verify stars are animating
- [ ] Switch to another tab - verify animations pause (check DevTools Performance)
- [ ] Switch back to tab - verify animations resume
- [ ] Scroll page - verify parallax effects work smoothly
- [ ] Check CPU usage in DevTools - should drop when tab is hidden

### Lazy Loading
- [ ] Open DevTools Network tab
- [ ] Load homepage - verify globe chunks are NOT loaded initially
- [ ] Scroll to map section - verify globe chunks load on-demand
- [ ] Verify IPFS images lazy load as you scroll
- [ ] Check that tooltips only load when hovered

### Web Vitals Monitoring
- [ ] Open browser console (development mode)
- [ ] Navigate through site - verify Web Vitals logs appear
- [ ] Check for any "Long task detected" warnings
- [ ] Verify no performance regressions in console

### Cross-Browser Testing
- [ ] Chrome/Edge - All features work
- [ ] Firefox - All features work
- [ ] Safari - All features work (especially 3D components)
- [ ] Mobile Safari - Performance is acceptable
- [ ] Mobile Chrome - Performance is acceptable

### Bundle Analysis
- [ ] Run `yarn analyze` locally
- [ ] Verify separate chunks for:
  - `thirdweb` (should be ~200-300KB)
  - `privy` (should be ~150-250KB)
  - `web3` (ethers/viem/wagmi)
  - `globe` (Three.js, only loaded async)
- [ ] Verify no duplicate dependencies in multiple chunks

### Edge Cases
- [ ] Slow 3G network throttling - verify graceful loading
- [ ] Disable JavaScript - verify SSR content appears
- [ ] Test with ad blockers - verify site still functions
- [ ] Test with browser extensions that modify DOM
- [ ] Clear cache and test cold load performance

### CODE REVIEW

File: ui/components/layout/SpaceBackground.tsx [hunk 1]
```diff
@@ -286,57 +307,81 @@ export default function SpaceBackground() {
     )
   }
 
+  // Page Visibility API - pause animations when tab is hidden
```
Comment (jaderiverstokes) [inline ui/components/layout/SpaceBackground.tsx]: maybe we can pull this out to it's own file. seems like it's repeated a bunch of times

**Review by jaderiverstokes — APPROVED** [review]
